docker/Dockerfile

```
@@ -1,13 +1,23 @@
-FROM alpine:3.5
+# build the server binary
+FROM golang:1.10.0 AS builder
+LABEL stage=intermediate
+WORKDIR /go/src/github.com/infobloxopen/atlas-pubsub
+COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -o pubsub ./cmd
 
-COPY bin/pubsub /usr/local/bin/pubsub
+
+# run the server binary
+FROM alpine:latest AS runner
+WORKDIR /usr/local/bin
+
+COPY --from=builder /go/src/github.com/infobloxopen/atlas-pubsub/pubsub .
 
 # Allows to verify certificates
 RUN apk update --no-cache && apk add --no-cache ca-certificates
```

```suggestion
 RUN apk add --no-cache ca-certificates
```

`apk update --no-cache` doesn't do anything, since it doesn't cache the files it just updated.

---

I'm a little confused by your comment. Didn't know too much about it so I have a resource here:
https://wiki.alpinelinux.org/wiki/Alpine_Linux_package_management#Update_the_Package_list

TLDR: It says it does cache files and that it's good to run updates before adds.

---

Sorry, I meant to reference the `--no-cache` documentation, but I forgot to paste.

https://github.com/gliderlabs/docker-alpine/blob/master/docs/usage.md#disabling-cache

`--no-cache` implies `update` and deletes the repo data (what update populates) when it's done.